src/signals/incident-management/containers/IncidentDetail/IncidentDetail.tsx

```diff
@@ -71,24 +75,44 @@ const Preview = styled.div`
 const IncidentDetail = () => {
   const { emit, listenFor, unlisten } = useEventEmitter()
   const storeDispatch = useDispatch()
-  const { id } = useParams()
+  const { id } = useParams<{ id: string }>()
   const [state, dispatch] = useReducer(reducer, initialState)
   const {
     error,
     get: getIncident,
     data: incident,
     isSuccess,
     patch,
-  } = useFetch()
-  const { get: getHistory, data: history } = useFetch()
-  const { get: getAttachments, data: attachments } = useFetch()
-  const { get: getDefaultTexts, data: defaultTexts } = useFetch()
-  const { get: getChildren, data: children } = useFetch()
-  const { get: getChildrenHistory, data: childrenHistory } = useFetchAll()
-  const { get: getContext, data: context } = useFetch()
-  const { get: getChildIncidents, data: childIncidents } = useFetchAll()
+  } = useFetch<Incident>()
+  const { get: getHistory, data: history } = useFetch<HistoryEntry[]>()
+  const { get: getAttachments, data: attachments } =
+    useFetch<Result<Attachment>>()
+  const { get: getDefaultTexts, data: defaultTexts } = useFetch<DefaultTexts>()
+  const { get: getChildren, data: children } = useFetch<Result<IncidentChild>>()
+  const { get: getChildrenHistory, data: childrenHistory } =
+    useFetchAll<HistoryEntry[]>()
+  const { get: getContext, data: context } = useFetch<Context>()
+  const { get: getChildIncidents, data: childIncidents } =
+    useFetchAll<Incident>()
 
   const subcategories = useSelector(makeSelectSubCategories)
+  const closeDispatch = () => dispatch({ type: CLOSE_ALL })
+
+  type KeyUpEvent = {
```

Comment:
is dit niet gewoon het generieke KeyboardEvent?

Reply:
dat heb ik in eerste instantie geprobeerd, maar dat lukte mij niet om goed te krijgen. Daarom heb ik het zo opgelost...

Reply:
maar nu wel, gek genoeg. ik heb het aangepast